### PR TITLE
Document advertising additional startd attributes (SOFTWARE-5324)

### PR DESCRIPTION
**Pull request by matyasselmeci**
This is one of those "if you don't know why you'd want to use it, you probably shouldn't" features but there isn't really a place for reference documentation and I don't want to have it be undocumented, so here it is.

### CODE REVIEW

**Comment by brianhlin** [comment]
Since @osg-cat has been talking about how he wants to rework this doc, I'd like to leave it up to him to determine whether or not this should go here. I think I'd prefer reference docs to go into the [README](https://github.com/opensciencegrid/osgvo-docker-pilot#readme)

**Comment by osg-cat** [comment]
I’m fine adding this for now, so that it doesn’t get lost.

File: docs/resource-sharing/os-backfill-containers.md [hunk 1]
```diff
@@ -213,6 +213,26 @@ the `--memory` argument tells Docker to kill the container if its total memory u
     Additionally, the processes that manage jobs also use some amount of memory.
     Therefore, it is important to give the container some extra room.
 
+
+### Advanced: Advertising additional pilot attributes
+
+You can put arbitrary additional attributes in the machine ads that you advertise to the OSPool.
```
Comment (brianhlin) [inline docs/resource-sharing/os-backfill-containers.md:219]:
```suggestion
You can put arbitrary additional attributes in the machine ads that the glidein advertises to the OSPool.
```

File: docs/resource-sharing/os-backfill-containers.md [hunk 2]
```diff
@@ -213,6 +213,26 @@ the `--memory` argument tells Docker to kill the container if its total memory u
...
+These attributes will show up when you do `condor_status -l` on your machine.
```
Comment (brianhlin) [inline docs/resource-sharing/os-backfill-containers.md:220]:
```suggestion
These attributes will show up when users run `condor_status -l` against your glidein.
```

I think it's safe to assume that some of these sites don't have HTCondor client tools

File: docs/resource-sharing/os-backfill-containers.md [hunk 3]
```diff
@@ -213,6 +213,26 @@ the `--memory` argument tells Docker to kill the container if its total memory u
...
+This could be useful for advertising something about the way the pilot was provisioned.
+To do this, volume-mount a file containing `key=value` pairs to `/etc/osg/extra-attributes.cfg`.
+Keys must be valid classad attribute names and values must be valid classad expressions.
+Multi-line strings are not supported.
+A line starting with `#` will be treated as a comment.
+For example:
+
+```
+# The Kubernetes namespace this pod is running under
+Kubernetes_Namespace = "path-osgdev"
```
Comment (brianhlin) [inline docs/resource-sharing/os-backfill-containers.md:230]:
```suggestion
KUBERNETES_NAMESPACE = "path-osgdev"
```

Unless there's some convention for these attrs that indicate that we should use title casing

File: docs/resource-sharing/os-backfill-containers.md [hunk 4]
```diff
@@ -213,6 +213,26 @@ the `--memory` argument tells Docker to kill the container if its total memory u
...
+# The deployment for this pilot
+Kubernetes_Deployment = "osgvo-docker-pilot-gpu"
```
Comment (brianhlin) [inline docs/resource-sharing/os-backfill-containers.md:232]:
```suggestion
KUBERNETES_DEPLOYMENT = "osgvo-docker-pilot-gpu"
```

**Review by brianhlin — APPROVED** [review]
Some minor comments. Pre-approving

File: docs/resource-sharing/os-backfill-containers.md (hunk 1 above)
Comment (matyasselmeci) [reply docs/resource-sharing/os-backfill-containers.md:219]: We don't call them glideins unless they're submitted by a factory but I can use "the pilot advertises" instead

File: docs/resource-sharing/os-backfill-containers.md (hunk 2 above)
Comment (matyasselmeci) [inline docs/resource-sharing/os-backfill-containers.md:220]:
```suggestion
These attributes will show up when users run `condor_status -l` against your pilot.
```

File: docs/resource-sharing/os-backfill-containers.md (hunk 1 above)
Comment (matyasselmeci) [inline docs/resource-sharing/os-backfill-containers.md:219]:
```suggestion
You can put arbitrary additional attributes in the machine ads that the pilot advertises to the OSPool.
```

**Pull request merged**